pkg/controller/azure/provider/resourcegroup.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2018 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provider
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/crossplaneio/crossplane/pkg/apis/azure/v1alpha1"
+	corev1alpha1 "github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane/pkg/clients/azure/resourcegroup"
+	"github.com/crossplaneio/crossplane/pkg/logging"
+	"github.com/crossplaneio/crossplane/pkg/util"
+)
+
+const (
+	controllerNameRG = "azure.resourcegroup"
+	finalizerRG      = "finalizer." + controllerNameRG
+
+	reasonFetchingClient   = "failed to fetch Azure Resource Group client"
+	reasonCreatingResource = "failed to create Azure Resource Group resource"
+	reasonDeletingResource = "failed to delete Azure Resource Group resource"
+	reasonSyncingResource  = "failed to sync Azure Resource Group resource"
+
+	azureDeletedMessage = "resource has been deleted on Azure"
+
+	reconcileTimeout = 1 * time.Minute
+)
+
+var logRG = logging.Logger.WithName("controller." + controllerNameRG)
```

Comment:
Yep, it may be difficult to discern what is happening with the Resource Groups when looking at logs if we remove this. Happy to do either, but my inclination would be to keep for now or find another workaround.

Reply:
Yeah, I think this is unfortunate but don't want to block getting this merged while we figure out a better pattern.